GeeksCoreLibrary/Modules/Databases/Services/MySqlDatabaseConnection.cs

```diff
@@ -139,32 +138,30 @@ private async Task<DataTable> GetAsync(string query, int retryCount, bool cleanU
             }
             catch (MySqlException mySqlException)
             {
-                if (retryCount >= gclSettings.MaximumRetryCountForQueries)
+                // Never retry single queries if we're in a transaction, because transactions will get rolled back when a deadlock occurs,
+                // so retrying a single query in a transaction is not very useful on most/all cases.
+                // Also, if we've reached the maximum number of retries, don't retry anymore.
+                if (HasActiveTransaction() || retryCount >= gclSettings.MaximumRetryCountForQueries)
                 {
-                    logger.LogError(mySqlException, "Error trying to run this query: {query}", query);
+                    logger.LogError(mySqlException, $"Error trying to run this query: {query}", query);
```

Comment:
Replacement werd al via parameter gedaan en is door dollar teken nu overbodig. Graag één van de twee toepassen bij alle soortgelijken.